keps/sig-auth/3221-multiple-authorization-webhooks/README.md

```diff
@@ -0,0 +1,623 @@
+
+
+### Test Plan
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+##### Unit tests
+
+Unit tests would be added along with any new code introduced.
+
+Existing test coverage:
+- cmd/kube-apiserver/app/options: 65
+- staging/src/k8s.io/apiserver/pkg/authorization: NA
+- staging/src/k8s.io/apiserver/apis/config/v1: 54.5
+
+> Will add more packages to the list during implementation.
+
+##### Integration tests
+
+Integration tests would be added to ensure the following:
+- Authorization of requests work in the existing flag based mode (feature flag
+turned off)
+- Authorization of requests work with an apiserver bootstrapped with
+authorization configuration file (feature flag turned on)
+    - without an webhook
+    - with an webhook - successful request
+    - with an webhook - error on request with `onError: Deny`
+    - with an webhook - error on request with `onError: NoOpinion`
+
+##### e2e tests
+
+End-to-end tests won't be needed as unit and integration tests would cover all
+the scenarios.
+
+### Graduation Criteria
+
+#### Alpha (1.25)
+
+- Add file based authorizer chain configuration
+- Add feature flag for gating usage
+- Unit tests and Integration tests to be written
+
+#### Future Alpha versions
+
+- Revisit on the items mentioned in Non-Goals and see if any needs to be implemented
+
+#### Beta
+
+- Address user reviews and iterate (if needed, keep in Alpha until changes stabilize)
+- Feature flag will be turned on by default
+
+#### GA
+
+- Feature flag removed
+- Existing command line flags will be marked as deprecated and config file will take
+precedence over the old flags
+
+#### GA + 3 cycles
+
+- Remove the existing command line flags
+
+### Upgrade / Downgrade Strategy
+
+While the feature is in Alpha, there is no change if cluster administrators want to
+keep on using command line flags.
+
+When the feature goes to Beta/GA or the cluster administrators want to configure
+authorizers using a config file, they need to make sure the config file exists before
+upgrading the cluster. Similarly when downgrading clusters, they would need to add
+the flags back to their bootstrap mechanism.
+
+### Version Skew Strategy
+
+Not applicable.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate
+  - Feature gate name: `AuthorizationConfigFromFile`
+  - Components depending on the feature gate:
+        - kube-apiserver
+
+###### Does enabling the feature change any default behavior?
+
+Yes, kube-apiserver will not use the `--authorization-*` to configure the authorizer
+chain. `kube-apiserver` will need the `--authorization-config-file` flag to be set to
+a valid config file on disk.
```

Comment:
Neither the old or the new flags is required IIRC.  So unless that changes, I am not sure if any default has actually changed.

Reply:
We will keep the old flags.